amundsen_application/static/js/components/common/EditableSection/index.tsx

```diff
@@ -114,11 +118,15 @@ export class EditableSection extends React.Component<
 
     return (
       <section className="editable-section">
-        <div className="section-title title-3">
-          {EditableSection.convertText(title)}
-          {!readOnly ? this.renderButton() : this.renderReadOnlyButton()}
-        </div>
-        {childrenWithProps}
+        <label className="editable-section-label">
+          <div className="editable-section-label-wrapper">
+            <span className="section-title title-3">
+              {EditableSection.convertText(title)}
+            </span>
+            {!readOnly ? this.renderButton() : this.renderReadOnlyButton()}
+          </div>
+          <div className="editable-section-content">{childrenWithProps}</div>
+        </label>
```

Comment:
This feature is something that a lot of people don't know.
Since the beginning, the labels were supposed to wrap the inputs they label, and only when they don't is when we use the "for" attribute. Somehow this knowledge got lost in the last years of FE development, but it was the original idea.

In this case, as the inputs of this could be different, it made sense to make the label wrap the input so that we don't need to keep a dynamic id on the inputs and add it on the label as the "for" attribute value. I had to add some extra markup to keep the right styling, but it is worth to keep on having this dynamic editable section.

Reply:
This makes sense.